ranges.md

```diff
@@ -0,0 +1,390 @@
+Versions.
+
+A Comparator is composed of an `Operator` and a `Version`.  The set
+of primitive Operators is:
+
+* `<` Less than
+* `<=` Less than or equal to
+* `>` Greater than
+* `>=` Greater than or equal to
+* `=` Equal.  If no operator is specified, then equality is assumed,
+  so this operator is optional, but MAY be included to differentiate it
+  from a SemVer Version string.
+
+A Comparator includes a given SemVer Version string if and only if the
+SemVer Version string has a precedence value greater than, less than, or
+equal to the Comparator, as appropriate to the Comparator's Operator.
+
+Additional Range forms are available for greater ergonomics, built upon
+these primitive Comparators.
+
+Introduction
+------------
+
+In a software library ecosystem, it is useful to define a specific range of
+versions which is known to satisfy a dependency.
+
+In an ideal world, where all publishers of in the ecosystem are following
+the SemVer guidelines, one could simply define a single "known-good"
+version of a dependency, and trust that any change to the `PATCH` or
+`MINOR` version would be acceptable without additional human attention.
+
+However, in practice, this is not always satisfactory.
+
+- A change to the internal workings of a library may not change the public
+  API, but nevertheless cause problems for users who have inadvertently
+  come to depend on subtle interactions.
+- A bug or security vulnerability introduced in a PATCH or MINOR update
+  might only end up getting fixed in a MAJOR update, and a downstream
+  consumer is unable to perform the necessary changes to adopt it.
+- Different consumers of a library may have different levels of risk
+  tolerance, with some preferring to adopt all new improvements, and others
+  opting to only upgrade after careful deliberation, even for PATCH and
+  MINOR updates.
+- PRERELEASE versions in particular are hazardous, and should only be used
+  when the consumer has explicitly opted into the risk involved in using
+  pre-release software.
+- SemVer is used for more than merely dependency resolution.  For example,
+  a library may wish to express the versions of a platform that it is
+  compatible with, and these might span multiple (but not all) MAJOR
+  version numbers.
+
+This specification aims to capture and formalize the version range
+semantics in popular usage within software ecosystems in the wild that have
+adopted Semantic Versioning.
+
+Semantic Version Range Specification (SemVer Ranges)
+----------------------------------------------------
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+The characters `I`, `J`, `K`, `L`, `M`, and `N` are all presumed to be
+positive integers without a leading zero, as allowed by the SemVer MAJOR,
+MINOR, and PATCH sections.
```

Comment:
the SemVer MAJOR/MINOR/PATCH sections allow for just `0`, which isn't technically positive (and since its a zero by itself, arguably its a leading zero?)

Reply:
Good catch.  I'll update to clarify.

Reply:
`0` by itself is not a leading zero. To be leading, it must have digits following it.

Reply:
`non-negative integer`, then?